modules/redis/redis.go

```diff
@@ -0,0 +1,267 @@
+package redis
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+
+	"github.com/go-redis/redis"
+	"github.com/netdata/go-orchestrator/module"
+)
+
+func init() {
+	creator := module.Creator{
+		Defaults: module.Defaults{
+			Disabled: false, // @TODO change it back
+		},
+		Create: func() module.Module { return New() },
+	}
+
+	module.Register("redis", creator)
+}
+
+const (
+	defaultHost = "localhost"
+	defaultPort = 6379
+)
+
+// New returns a pointer to a Redis instance with default values.
+func New() *Redis {
+	return &Redis{
+		charts: charts.Copy(),
+		Config: Config{
+			Host:     defaultHost,
+			Port:     defaultPort,
+			Password: "",
+			DbNum:    0,
+		},
+	}
+}
+
+// Config for the Redis module
+type Config struct {
+	Host     string `yaml:"host"`
+	Port     int    `yaml:"port"`
+	Password string `yaml:"password"`
+	DbNum    int    `yaml:"dbnum"`
+}
+
+// Redis module
+type Redis struct {
+	module.Base
+	charts      *Charts
+	client      *redis.Client
+	bgSaveTime  int64
+	Config      `yaml:",inline"`
+	UpdateEvery int `yaml:"update_every"`
+}
+
+// Cleanup closes the connection to the Redis server
+func (r *Redis) Cleanup() {
+	if err := r.client.Close(); err != nil {
+		r.Errorf("cleanup: error on closing the redis client [%+v]: %v", r.Config, err)
+	}
+}
+
+// Init checks the configuration parameters and pings the Redis server
+func (r *Redis) Init() bool {
+	if r.Host == "" {
+		r.Error("Host is not set")
+		return false
+	}
+
+	if r.Port == 0 {
+		r.Error("Port is not set")
+		return false
+	}
+
+	client := redis.NewClient(&redis.Options{
+		Addr:     r.Host + ":" + strconv.Itoa(r.Port),
+		Password: r.Password,
+		DB:       r.DbNum,
+	})
+
+	_, err := client.Ping().Result()
+	if err != nil {
+		r.Errorf("init: error on starting the redis client [%+v]: %v", r.Config, err)
+		return false
+	}
+
+	r.client = client
+	r.bgSaveTime = 0
+
+	r.Debugf("using config %+v", r.Config)
+
+	return true
+}
+
+// Check makes sure that at least one metric is being collected
+func (r *Redis) Check() bool {
+	return len(r.Collect()) > 0
+}
+
+// Charts returns the Redis charts
+func (r *Redis) Charts() *Charts {
+	return r.charts
+}
+
+// Collect returns a map of metrics
+func (r *Redis) Collect() map[string]int64 {
+	metrics := make(map[string]int64)
+
+	res, err := r.client.Do("INFO").Result()
+	if err != nil {
+		r.Errorf("Could not get Redis INFO: %v", err)
+		return nil
+	}
+
+	err = parseMetrics(res.(string), r.charts, metrics, &r.bgSaveTime, r.UpdateEvery)
+	if err != nil {
+		r.Errorf("Got error while parsing metrics: %v", err)
+		r.Debugf("INFO: %+v", res)
+	}
+
+	r.Debugf("Metrics: %+v", metrics)
+
+	return metrics
```

Comment:
Please move collect logic to `collect.go`. Reason - it is nice to have `collect.go` as an entry point for collecting logic for all modules.


https://github.com/netdata/go.d.plugin/blob/02b5f76a43e0f8cb69e936f3a33560797ae2b178/modules/zookeeper/zookeeper.go#L95-L105

Reply:
Sounds good to me :+1: